examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+  // be implemented. It creates the actual triangulation that is a
+  // benchmark configuration consisting of a channel with a disc obstacle, see
+  // @cite GuermondEtAl2018. We construct the geometry by modifying the
+  // mesh generated by GridGenerator::hyper_cube_with_cylindrical_hole().
+  // We refer to Step-49, Step-53, and Step-54 for an overview how to
+  // create advanced meshes.
+
+  template <int dim>
+  void Discretization<dim>::setup()
+  {
+    TimerOutput::Scope t(computing_timer, "discretization - setup");
+
+    triangulation.clear();
+
+    // We first create 4 temporary (non distributed) coarse triangulations
+    // that we stitch together with the
+    // GridGenerator::merge_triangulation() function. We center the disc at
+    // $(0,0)$ with a diameter of <code>disc_diameter</code>. The lower
+    // left corner of the channel has coordinates
+    // (<code>-disc_position</code>, <code>-height/2</code>) and the upper
+    // right corner has (<code>length-disc_position</code>,
+    // <code>height/2</code>).
+
+    Triangulation<dim> tria1, tria2, tria3, tria4;
+
+    GridGenerator::hyper_cube_with_cylindrical_hole(
+      tria1, disc_diameter / 2., disc_diameter, 0.5, 1, false);
+
+    GridGenerator::subdivided_hyper_rectangle(
+      tria2,
+      {2, 1},
+      Point<2>(-disc_diameter, disc_diameter),
+      Point<2>(disc_diameter, height / 2.));
+
+    GridGenerator::subdivided_hyper_rectangle(
+      tria3,
+      {2, 1},
+      Point<2>(-disc_diameter, -disc_diameter),
+      Point<2>(disc_diameter, -height / 2.));
+
+    GridGenerator::subdivided_hyper_rectangle(
+      tria4,
+      {6, 4},
+      Point<2>(disc_diameter, -height / 2.),
+      Point<2>(length - disc_position, height / 2.));
+
+    GridGenerator::merge_triangulations({&tria1, &tria2, &tria3, &tria4},
+                                        triangulation,
+                                        1.e-12,
+                                        true);
+
+    triangulation.set_manifold(0, PolarManifold<2>(Point<2>()));
+
+    // We have to fix up the left edge that is currently located at
+    // $x=-$<code>disc_diameter</code> and has to be shifted to
+    // $x=-$<code>disc_position</code>:
+
+    for (auto cell : triangulation.active_cell_iterators())
+      for (unsigned int v = 0; v < GeometryInfo<dim>::vertices_per_cell; ++v)
+        {
+          auto &vertex = cell->vertex(v);
+          if (vertex[0] <= -disc_diameter + 1.e-6)
+            vertex[0] = -disc_position;
+        }
+
+    // As a last step the boundary has to be colorized with
+    // <code>Boundary::do_nothing</code> on the right,
+    // <code>Boundary::dirichlet</code> on the left and
+    // <code>Boundary::slip</code> on the upper and lower outer boundaries
+    // and the obstacle:
+
+    for (auto cell : triangulation.active_cell_iterators())
+      {
+        for (unsigned int f = 0; f < GeometryInfo<2>::faces_per_cell; ++f)
```

Comment:
Remove the single open brace in the line above the loop to make this nested loop look like the previous one. Separately, here you use `GeometryInfo<2>`, but above `GeometryInfo<dim>`. It would be good to have a common style.

Reply:
It might also be nice to just write this as 
```suggestion
        for (const auto &face : cell->face_iterators())
```